lib/hawk/model/association.rb

```diff
@@ -73,17 +73,16 @@ module ClassMethods
         def inherited(subclass)
           super
 
-          parent = self
           subclass.instance_eval do
             # Inherit associations
             @_associations ||= {}
 
-            parent.associations.each do |name, (type, options)|
+            self.associations.each do |name, (type, options)|
               _define_association(name, type, options)
```

Comment:
Here, as discussed, we should take care of mangling the `options[:class_name]` if we detect being in a namespace different than the one we are inheriting from.

THANKS